src/Coordination/CoordinationSettings.cpp

```diff
@@ -225,6 +225,15 @@ String KeeperConfigurationAndSettings::getSnapshotsPathFromConfig(const Poco::Ut
 
 String KeeperConfigurationAndSettings::getStateFilePathFromConfig(const Poco::Util::AbstractConfiguration & config, bool standalone_keeper_)
 {
+    if (config.has("keeper_server.storage_path"))
+        return std::filesystem::path{config.getString("keeper_server.storage_path")} / "state";
+
+    if (config.has("keeper_server.snapshot_storage_path"))
+        return std::filesystem::path(config.getString("keeper_server.snapshot_storage_path")).parent_path() / "state";
+
+    if (config.has("keeper_server.log_storage_path"))
+        return std::filesystem::path(config.getString("keeper_server.log_storage_path")).parent_path() / "state";
```

Comment:
Can we do it without this? As a user, it's hard to say where the file will end up (e.g. why is snapshot parent folder and not log parent folder if both are defined).

I completely missed `storage_path`, would that be enough?

Reply:
Usually they are the same I guess.